src/components/dds/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { componentClasses } from 'dds-core/components/pagination/pagination';
+import Select from '../Select';
+import type { SelectProps } from '../Select';
+import { uniqueId } from 'lodash';
+import { Button } from '../Button';
+import classNames from 'classnames';
+import PaginationRangeInput from './PaginationRangeInput';
+import type { PaginationRangeInputProps } from './PaginationRangeInput';
+import usePaginationSizeClassName from './usePaginationSizeClassName';
+import { getPagesCount, getPageSizeOptions } from './Pagination.utils';
+import { matchers } from 'dds-core/js/validate-props';
+
+const firstPage = 1;
+
+export interface DDSPaginationProps {
+    /**
+     * Total number of available items
+     */
+    totalItems: number;
+    /**
+     * Number of items displayed per page
+     */
+    pageSize?: number;
+    /**
+     * Function called when selected page or number of items per page has changed
+     */
+    onChange: (page: number, pageSize: number) => void;
+}
+
+/**
+ * React implementation of DDS pagination component
+ * The component name is temporarily prefixed with 'DDS' due to name conflict with SemanticUI-based `Pagination` component.
+ */
+const DDSPagination = ({ totalItems, pageSize: initialPageSize = 5, onChange }: DDSPaginationProps) => {
+    const paginationPerPageId = uniqueId();
+    const pageRangeId = uniqueId();
+    const paginationRef = useRef<HTMLDivElement | null>(null);
+    const resizableClassName = usePaginationSizeClassName(paginationRef);
+
+    const [currentPage, setCurrentPage] = useState(firstPage);
+    const [pageSize, setPageSize] = useState(initialPageSize);
+    const lastPage = getPagesCount(totalItems, pageSize);
+    const itemsPerPageOptions = getPageSizeOptions(totalItems, initialPageSize);
```

Comment:
Maybe
```suggestion
    const pageSizeOptions = getPageSizeOptions(totalItems, initialPageSize);
```
to be fully inline?

Reply:
Sure, seems like a nice idea 💪 
Applied as a part of 7aecc780e29b49131f58121760bfc641c4ca9ef5